script.js

```diff
@@ -60,10 +60,17 @@ function storeNonEmptyCurrentDisplay() {
 
 
 function evaluateNow() {
-    if (numberArray.length === 1) ans = numberArray[0];
+    if (numberArray.length === 1 && !operatorSet) ans = numberArray[0];
     else {
-        const last = numberArray[numberArray.length-1];
-        const secondLast = numberArray[numberArray.length-2];
+        let last;
+        let secondLast;
+        if (operatorSet) {
+          last = lastNumber;
+          secondLast = numberArray.pop();
+        } else {
+          last = numberArray.pop();
+          secondLast = numberArray.pop();
```

Comment:
Replacing the evaluateNow() and evaluateEverythingNow() method of appending answer to numberArray while keeping unused numbers intact